src/main/java/bio/terra/common/LockOperation.java

```diff
@@ -0,0 +1,33 @@
+package bio.terra.common;
+
+import java.util.List;
+
+/** Actions which could be taken to modify the locks held on a resource (dataset, snapshot). */
+public enum LockOperation {
+  LockExclusive,
+  LockShared,
+  UnlockExclusive,
+  UnlockShared;
```

Comment:
Since enum fields are static final data the usual convention is to name them as other java constants:
```suggestion
  LOCK_EXCLUSIVE,
  LOCK_SHARED,
  UNLOCK_EXCLUSIVE,
  UNLOCK_SHARED;
```